src/services/moveToFileAndNewFile.ts

```diff
@@ -0,0 +1,795 @@
+import { cast, concatenate, contains, emptyArray, find, findIndex, firstDefined, flatMap, GetCanonicalFileName, getRangesWhere, last,length,mapDefined,some,tryCast } from "../compiler/core";

```

Comment:
Nit: I’m not a fan of this file name, and since `moveToFile` handles moving both to an existing file and to a new file, `moveToNewFile` now feels like just a special case of `moveToFile`. I think it would be ok to just put everything in `moveToFile.ts` and let `moveToNewFile.ts` import directly from `moveToFile.ts`.

Reply:
Yes, if you do what Andrew said, then you don't need all of those `@internal` comments either.